packages/2d/src/components/Line.ts

```diff
@@ -265,15 +265,25 @@ export class Line extends Shape {
     context.beginPath();
     if (this.endArrow() instanceof Node) {
       const endArrow = this.endArrow() as Node;
-      endArrow.position(endPoint);
+      context.translate(endPoint.x, endPoint.y);
```

Comment:
We also need to find a way to scale the arrows down when there's not enough space on the line.
Otherwise animating the `start/end` parameters will be cumbersome.

Reply:
could you elaborate on that?

Reply:
Try running the following example:
```ts
import {makeScene2D} from '@motion-canvas/2d';
import {Line} from '@motion-canvas/2d/lib/components';
import {all, waitFor} from '@motion-canvas/core/lib/flow';
import {createRef} from '@motion-canvas/core/lib/utils';

export default makeScene2D(function* (view) {
  const line = createRef<Line>();

  view.add(
    <Line
      ref={line}
      points={[
        [-400, 0],
        [400, 0],
      ]}
      lineWidth={8}
      stroke={'white'}
      startArrow
      endArrow
      arrowSize={100}
    />,
  );

  yield* all(
    line().start(1, 3),
    line().end(0, 3),
  );

  yield* waitFor(5);
});
```
Description | Image
--- | ---
I exaggerated the size to better illustrate the effect. | ![image](https://user-images.githubusercontent.com/64662184/220876559-3a8d342d-50a0-425a-99a1-5c14ed396f36.png)
Notice that when the line passes through the point of zero length the arrows get scaled down | ![image](https://user-images.githubusercontent.com/64662184/220876509-e7f2881c-6d8c-4cc3-8a19-dc7d0c678b10.png)
Without that, it results in this ugly effect: | ![image](https://user-images.githubusercontent.com/64662184/220876465-b41363dc-2d0a-43be-8256-61bf6e4cdecb.png)

This is done automatically so that you don't need to animate `arrowSize` each time this happens.

Reply:
Should we do this by scaling or "cutting off" the node, if it overlaps?

Reply:
I'm really sorry, I didn't notice your comment.
I think scaling down the node would be better because it would work even if the arrow exceeds the actual line (for example a circle whose center is placed at the end of the line).
You should be able to use `Node.cacheBBox()` to get the size of the node so that you know when to start scaling it down.